modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java

```diff
@@ -125,8 +131,11 @@ protected DDMFormRenderingContext createDDMFormRenderingContext(
 		DDMFormRenderingContext formRenderingContext =
 			new DDMFormRenderingContext();
 
-		formRenderingContext.setHttpServletRequest(
-			_ddmFormAdminRequestHelper.getRequest());
+		HttpServletRequest httpServletRequest =
+			_ddmFormAdminRequestHelper.getRequest();
+
+		formRenderingContext.setHttpServletRequest(httpServletRequest);
+
```

Comment:
@marcosapmf, why was is necessary?

Reply:
it's necessary to avoid a NPE [here](https://github.com/liferay/liferay-portal/blob/2152cecbba72d32572fa776486d7ea59bb0ce585/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormTemplateContextFactoryImpl.java#L126)

Reply:
To avoid the NPE then the variable `httpServletRequest` should be used there or some check should be performed there too, right? Can you please give more details, @marcosapmf? Thanks.